tdvt/test/tdvt_test.py

```diff
@@ -224,21 +224,45 @@ def build_tabquery_command_line_local(work):
     return new_cmd
 
 class CommandLineTest(unittest.TestCase):
-    def test_command_line_full(self):
-        test_config = TdvtTestConfig()
-        test_config.logical = False
-        test_config.tds = 'mytds.tds'
-        #Optional.
-        test_config.output_dir = 'my/output/dir'
-        test_config.d_override = '-DLogLevel=Debug'
-
-        test_file = 'some/test/file.txt'
-        test_set = ExpressionTestSet(TEST_DIRECTORY, 'mytest', test_config.tds, '', test_file, '')
-        work = tdvt_core.BatchQueueWork(test_config, test_set)
-        cmd_line = build_tabquery_command_line_local(work)
-        cmd_line_str = ' '.join(cmd_line)
-        expected = 'tabquerytool.exe --expression-file-list my/output/dir\mytest\\tests.txt -d mytds.tds --combined --output-dir my/output/dir -DLogDir=my/output/dir\mytest -DOverride=ProtocolServerNewLog -DLogLevel=Debug -DLogicalQueryRewriteDisable=Funcall:RewriteConstantFuncall'
-        self.assertTrue(cmd_line_str == expected, 'Actual: ' + cmd_line_str + ': Expected: ' + expected)
+    def test_command_line_full_windows(self):
+        if sys.platform in ('win32', 'cygwin'):
+            test_config = TdvtTestConfig()
+            test_config.logical = False
+            test_config.tds = 'mytds.tds'
+            #Optional.
+            test_config.output_dir = 'my/output/dir'
+            test_config.d_override = '-DLogLevel=Debug'
+
+            test_file = 'some/test/file.txt'
+            test_set = ExpressionTestSet(TEST_DIRECTORY, 'mytest', test_config.tds, '', test_file, '')
+            work = tdvt_core.BatchQueueWork(test_config, test_set)
+            cmd_line = build_tabquery_command_line_local(work)
+            cmd_line_str = ' '.join(cmd_line)
+            expected = 'tabquerytool.exe --expression-file-list my/output/dir\mytest\\tests.txt -d mytds.tds --combined --output-dir my/output/dir -DLogDir=my/output/dir\mytest -DOverride=ProtocolServerNewLog -DLogLevel=Debug -DLogicalQueryRewriteDisable=Funcall:RewriteConstantFuncall'
+            self.assertTrue(cmd_line_str == expected, 'Actual: ' + cmd_line_str + ': Expected: ' + expected)
+        else:
+            self.skipTest(reason="Not running on Windows.")
+
+    def test_command_line_full_mac_linux(self):
+        if sys.platform in ('darwin') or 'linux' in sys.platform:
```

Comment:
Update expected string based on platform, pull out remaining logic into another function to avoid duplicated code.  I didn't look closely, but similar comment may apply below as well.

Reply:
@rosswbrown I added a fixture to that test class and put a conditional in the tests to either use the correct `expected` value or skip the test if python thinks it is being run on an unsupported os.